runelite-client/src/main/java/net/runelite/client/plugins/xpglobes/XpGlobesOverlay.java

```diff
@@ -296,4 +310,62 @@ private void drawTooltip(Graphics2D graphics, XpGlobe mouseOverSkill, int goalXp
 
 		xpTooltip.render(graphics);
 	}
+
+	private void drawLevels(Graphics2D graphics, int startXp, int goalXp, int x, int y)
+	{
+		final Collection<Double> levelPercentages = levelPercentagesCache.computeIfAbsent(
+			Pair.of(startXp, goalXp),
+			unused -> Experience.levelPercentagesBetweenExperience(startXp, goalXp)
+		);
+
+
+		final Stroke backupStroke = graphics.getStroke();
+		graphics.setStroke(new BasicStroke(2));
+
+		for (final double levelMarker : levelPercentages)
+		{
+			final double angle = Math.toRadians(levelMarker * 360);
+			final Vector4<Double> coordinates = calculateCoordinatesForAngle(angle, x, y);
```

Comment:
Just draw your line instead of returning a new class that gets drawn

Reply:
I don't agree here. A function should do just do one thing so it can be easily reused, tested, and it's intention is clear. (See: clean code)
When Project Valhalla hits, the vector class will be inlined anyway.
Also, the performance overhead of creating and returning the class is insignificant either way. 99% of the time is spent on AWT either way.

Reply:
1. This function isn't going to be reused 
2. We are on Java 8 
3. You are still creating unnecessary code and abstracting logic where it doesn't need be abstracted